### ISSUE TITLE
Takes too long

### ISSUE THREAD

**Issue by ghost** [mikeyy/nonoCAPTCHA#24]
The custom NonoCAPTCHA tab times out everytime whereas just opening a new tab to the same link works perfectly.

**Comment by mikeyy** [comment]
I will need some more information to see if I can reproduce this. What operating system are you using? And is this with proxies?

**Comment by BenBaltz** [comment]
Windows. This was a run without proxies, I changed deface.html a bit to finally get it to work (added a grecaptcha.render) but was met with status detected every attempt. Then tried proxies, still status detected.

**Comment by mikeyy** [comment]
Unfortunately it doesn't get around being detected if your IP is marked as bad. I'm still looking into other possible ways they may be detecting us, such as a bad user-agent or if it's just a bad proxy in general. It's an ongoing process that is quite tedious considering I need to find a non-marked IP/proxy and hopefully get it to be detected and then non-detected on another go-around and see what the differences were.

I will take your method of adding a grecaptcha.render as an extra fallback incase the page fails to load. Sometimes initiating a page load event was required from Selenium to get the reCAPTCHA widget to load so it might be the same problem.

**Comment by BenBaltz** [comment]
I pay for smartproxy.io which is very high quality residentials, I also hotspotted from my phone same result. I don't think it likes grecaptcha.render. I tried removing the deface altogether but it would error out.

**Comment by mikeyy** [comment]
Could you try cloning the current branch on here and seeing if the code on line 156 works for you instead of grecaptcha.render?
```python
var evt = document.createEvent('Event');
evt.initEvent('load', false, false);
window.dispatchEvent(evt);
```

If the page still fails to load then I will set-up a virtual machine with Windows later and do some testing there. In the mean-time I'm still doing more testing on possible ways for being detected that haven't yet been released on PyPI. Hoping we can figure this out and get it working for you.

**Comment by BenBaltz** [comment]
Yeah I'll test it out thanks. If I cant would you be inclined to sell this as a service?

**Comment by mikeyy** [comment]
A service is in the works for the near future. 👍

**Comment by BenBaltz** [comment]
I'd suggest taking this down then or keeping the secret sauce off git if u know what I mean

**Comment by mikeyy** [comment]
Will keep that in mind, thanks for your consideration.

**Comment by mikeyy** [comment]
This issue just appeared on my end while working with the reCAPTCHA's demo page, and I have reason to believe it was due to Content-Security-Policy. It didn't always happen though so I must have disregarded it. I have pushed a patch to the branch and will release it to PyPI sometime today or later this week. Considering most issues have been worked out, I've been slow on making any releases unless changes are discernible or to fix a nasty bug that breaks the library.

**Comment by mikeyy** [comment]
The issue seems resolved https://github.com/miyakogi/pyppeteer/issues/121

**Comment by mikeyy** [comment]
Whoops, wrong issue...

**Comment by BenBaltz** [comment]
Dang it. I thought it was fixed

**Comment by mikeyy** [comment]
Actually, could you please give the latest release a try? It includes the bypass that enables scripts to execute on websites with an active Content-Security-Policy response header. I happened upon the issue when trying to deface the reCAPTCHA demo page and the reCAPTCHA widget wouldn't load. Hope this fixes it!
